src/welcome-bare-metal.md

```diff
@@ -0,0 +1,32 @@
+# Welcome to Bare Metal Rust
+
+This is a standalone one-day course about bare-metal Rust, aimed at people who are familiar with the
+basics of Rust (perhaps from completing the Comprehensive Rust course), and ideally also have some
+experience with bare-metal programming in some other language such as C.
+
+Today we will talk about 'bare-metal' Rust: running Rust code without an OS underneath us. This will
+be divided into several parts:
+
+ * What is `no_std` Rust?
+ * Writing firmware for microcontrollers.
+ * Writing bootloader / kernel code for application processors.
+ * Some useful crates for bare-metal Rust development.
+
+For the microcontroller part of the course we will use the [BBC micro:bit](https://microbit.org/) v2
+as an example. It's a [development board](https://tech.microbit.org/hardware/) based on the Nordic
+nRF51822 microcontroller with some LEDs and buttons, an I2C-connected accelerometer and compass, and
+an on-board SWD debugger.
+
+To get started, install some tools we'll need later:
+
+```bash
+sudo apt install gdb-multiarch picocom
+rustup update
+rustup target add thumbv7em-none-eabihf
+cargo install cargo-binutils cargo-embed
+echo 'SUBSYSTEM=="usb", ATTR{idVendor}=="0d28", MODE="0664", GROUP="plugdev"' | sudo tee /etc/udev/rules.d/50-microbit.rules
```

Comment:
I know the line is not meant to be read carefully, but it's too long to see without scrolling.

Many people will probably have questions about what this does, especially since you're asking them to execute it as root.

Reply:
Well okay, they are welcome to ask questions.

Reply:
Yes, I would like to move this command to it's own code block, with some text in front of it which explains why we're running commands as root. You'll want to explain to people what the command does and why it's needed -- that should be helpful for people who don't use `udev` or who are on a system where the config files are somewhere else.

The command should then be nicely formatted so that they're readable without overflow.

Reply:
Done.